src/modules/job-exec/job-exec.c

```diff
@@ -1505,6 +1505,11 @@ static void config_reload_cb (flux_t *h,
         }
         i++;
     }
+    if (flux_set_conf (h, flux_conf_incref (conf)) < 0) {
```

Comment:
Ah, heh! You did this in the midst of my review. Comments about combining config_init() and config_reconfig() still apply perhaps.